modules/gapi/src/api/gproto.cpp

```diff
@@ -197,6 +197,45 @@ bool cv::can_describe(const GMetaArgs &metas, const GRunArgs &args)
                      });
 }
 
+bool cv::validate_input_arg(const GRunArg& arg)
+{
+    // FIXME: It checks only Mat's argument
+    switch (arg.index())
+    {
+#if !defined(GAPI_STANDALONE)
+    case GRunArg::index_of<cv::Mat>():
+    {
+        const auto desc = descr_of(util::get<cv::Mat>(arg));
+        GAPI_Assert(desc.size.height != 0 && desc.size.width != 0 && "incorrect dimensions of cv::Mat!"); return true;
```

Comment:
you don't need `return` here. And please don't forget about `break` once you refactor

Reply:
Fixed